src/libs/actions/PersistedRequests.ts

```diff
@@ -31,18 +31,31 @@ function remove(requestToRemove: Request) {
      * We only remove the first matching request because the order of requests matters.
      * If we were to remove all matching requests, we can end up with a final state that is different than what the user intended.
      */
-    const requests = [...persistedRequests];
-    const index = requests.findIndex((persistedRequest) => isEqual(persistedRequest, requestToRemove));
-    if (index !== -1) {
-        requests.splice(index, 1);
+    const index = persistedRequests.findIndex((persistedRequest) => isEqual(persistedRequest, requestToRemove));
+    if (index === -1) {
+        return;
     }
+    persistedRequests.splice(index, 1);
+    Onyx.set(ONYXKEYS.PERSISTED_REQUESTS, persistedRequests);
+}
 
-    persistedRequests = requests;
-    Onyx.set(ONYXKEYS.PERSISTED_REQUESTS, requests);
+function update(oldRequest: number | Request, newRequest: Request) {
```

Comment:
Yeah, this is a bit weird in that I originally wrote this to take a request, but I later realized that we would already know the index of the request, so I updated it to take a request _or_ an index. That's why `oldRequest` is `number | Request`.

But this is unnecessarily complex I suppose and can be simplified to be just a number because that's the only way we use it now.

Reply:
done